bundles/AdminBundle/Resources/public/js/pimcore/document/tree.js

```diff
@@ -865,6 +865,27 @@ pimcore.document.tree = Class.create({
                 }
             }
 
+            // expand and collapse complete tree
+            if (record.data.type === "folder") {
```

Comment:
```suggestion
            if (!record.data.leaf) {
```
Also pages can have children 😉